xml-parser/src/test/java/org/uniprot/core/xml/uniref/UniRefEntryConverterTest.java

```diff
@@ -41,6 +48,69 @@ void testFromXml() throws Exception {
+                                        .taxonId(9606L)
+                                        .scientificName("Homo Sapiens")
+                                        .commonName("Human")
+                                        .build())
+                        .memberCount(1)
+                        .representativeMember(
+                                new RepresentativeMemberBuilder()
+                                        .memberId("P12345_HUMAN")
+                                        .memberIdType(UniRefMemberIdType.UNIPROTKB_TREMBL)
+                                        .accessionsAdd(
+                                                new UniProtKBAccessionBuilder("P12345").build())
+                                        .sequence(new SequenceBuilder("AAAAA").build())
+                                        .organismName("Homo Sapiens (Human)")
+                                        .organismTaxId(9606L)
+                                        .build())
+                        .build();
+        Entry xmlConverter = converter.toXml(uniRefEntry);
+        assertNotNull(xmlConverter);
+        UniRefEntry entryConverted = converter.fromXml(xmlConverter);
+        assertNotNull(entryConverted);
+        assertEquals(uniRefEntry, entryConverted);
+    }
+
+    @Test
+    void testCommonOrganismWithoutCommonName() throws Exception {
+        UniRefEntryConverter converter = new UniRefEntryConverter();
+        UniRefEntry uniRefEntry =
+                new UniRefEntryBuilder()
+                        .id("UniRef50_P12345")
+                        .entryType(UniRefType.UniRef50)
+                        .commonTaxon(
+                                new OrganismBuilder()
+                                        .taxonId(9606L)
+                                        .scientificName("Homo Sapiens")
+                                        .build())
+                        .memberCount(1)
+                        .representativeMember(
+                                new RepresentativeMemberBuilder()
+                                        .memberId("P12345_HUMAN")
+                                        .memberIdType(UniRefMemberIdType.UNIPROTKB_TREMBL)
+                                        .accessionsAdd(
+                                                new UniProtKBAccessionBuilder("P12345").build())
+                                        .sequence(new SequenceBuilder("AAAAA").build())
+                                        .organismName("Homo Sapiens")
+                                        .organismTaxId(9606L)
+                                        .build())
+                        .build();
+        Entry xmlConverter = converter.toXml(uniRefEntry);
+        assertNotNull(xmlConverter);
+        UniRefEntry entryConverted = converter.fromXml(xmlConverter);
+        assertNotNull(entryConverted);
+        assertEquals(uniRefEntry, entryConverted);
+    }
+
```

Comment:
Is it possible to have commanTaxon  without either taxonId  or  scientificName  or  both? If yes, we should cover that in our  test coverage.

Reply:
Leo one more thing, Aurelien has mentioned about making uniparcId as String field. I didn't see changes related to  that.  Are we not doing that?

Reply:
The change for uniparcId  is in the class: UniRefEntryLightJsonConfig.java,  line 98

Reply:
All good Leo. Please merge.